src/middleware/jira-jwt-middleware.ts

```diff
@@ -1,41 +1,104 @@
-import { Installation } from "models/installation";
+import { decodeSymmetric, getAlgorithm } from "atlassian-jwt";
 import { NextFunction, Request, Response } from "express";
-import { sendError, TokenType, verifySymmetricJwtTokenMiddleware } from "../jira/util/jwt";
+import { extractJwtFromRequest, TokenType, verifySymmetricJwtAndSetResponseCodeOnError } from "~/src/jira/util/jwt";
+import { Installation } from "~/src/models/installation";
 
-export const verifyJiraJwtMiddleware = (tokenType: TokenType) => async (
-	req: Request,
-	res: Response,
-	next: NextFunction
-): Promise<void> => {
-	const { jiraHost } = res.locals;
+export const jiraHostMiddleware = async (req: Request, res: Response, next: NextFunction) => {
 
-	if (!jiraHost) {
-		sendError(res, 401, "Unauthorised");
+	const token = extractJwtFromRequest(req);
+
+	if (!token) {
+		if (req.session.jiraHost) {
+			res.locals.jiraHost = req.session.jiraHost;
+			next();
+			return;
+		}
+		next();
 		return;
 	}
 
-	const installation = await Installation.getForHost(jiraHost);
-
+	const installation = await getInstallationAndTokenVerification(token, req, res);
 	if (!installation) {
-		return next(new Error("Not Found"));
+		return;
```

Comment:
Hmm, this just return but not calling `next?`

Reply:
`getInstallationAndTokenVerification` method sets proper msg on response object, so just return in case of any error.

Reply:
Since `getInstallationAndTokenVerification` is completely private to this file, can you refactor it a bit so that the return is more explicity (like we don't implicitly depends on it to call `res.status().send()`)

It could be error prone implicitly depends on it to do something. I'd prefer having it return more explicit result either success or fail with error code (or throw with error )